source/UF/H-Levels.lagda

```diff
@@ -234,3 +234,37 @@ From Univalence we can show that (ℍ n) is of level (n + 1), for all n : ℕ.
    e' = (pr₁ , pr₁-is-embedding (λ f → being-equiv-is-prop fe f))
 
 \end{code}
+
+We now define the notion of a k-truncation using record types.
+
+\begin{code}
+
+record H-level-truncations-exist : 𝓤ω where
+ field
+  ∣∣_∣∣_ : {𝓤 : Universe} → 𝓤 ̇ → ℕ → 𝓤 ̇
+  ∣∣∣∣-is-prop : {𝓤 : Universe} {X : 𝓤 ̇ } {n : ℕ} → is-prop (∣∣ X ∣∣ n)
+  ∣_∣_ :  {𝓤 : Universe} {X : 𝓤 ̇ } → X → (n : ℕ) → ∣∣ X ∣∣ n
+  ∣∣∣∣-rec : {𝓤 𝓥 : Universe} {X : 𝓤 ̇ } {Y : 𝓥 ̇ } {n : ℕ}
+           → Y is-of-hlevel n → (X → Y) → ∣∣ X ∣∣ n → Y
+ infix 0 ∣∣_∣∣_
+ infix 0 ∣_∣_
+
+\end{code}
+
+We now add the notion of k-connectedness of type and functions with respect to
```

Comment:
Maybe better to say "define" rather than "add"? Think as if you were writing a paper, rather than building a library.